tensorflow_addons/optimizers/weight_decay_optimizers.py

```diff
@@ -261,10 +261,17 @@ def _do_use_weight_decay(self, var):
         return var.ref() in self._decay_var_list
 
 
+optimizer_class = Union[
```

Comment:
Why not BaseOptimizerClass?

Reply:
I am renaming this to keras_legacy_optimizer to keep aligned with changes suggested by Michal.